src/main.cpp

```diff
@@ -13,10 +13,189 @@
 #include "hdf5_io.hpp"
 #include "ocl_dev_mgr.hpp"
 #include "timer.hpp"
+#include <chrono>
+#include <sstream> 
+#include <fstream>
+#include <time.h>
+
+
+#if defined(_WIN32)
+#include <windows.h>
+#include <sys/timeb.h>
+
+int gettimeofday(struct timeval * tp, struct timezone * tzp)
+{
+	static const uint64_t EPOCH = ((uint64_t)116444736000000000ULL);
+
+	SYSTEMTIME  system_time;
+	FILETIME    file_time;
+	uint64_t    time;
+
+	GetSystemTime(&system_time);
+	SystemTimeToFileTime(&system_time, &file_time);
+	time = ((uint64_t)file_time.dwLowDateTime);
+	time += ((uint64_t)file_time.dwHighDateTime) << 32;
+
+	tp->tv_sec = (long)((time - EPOCH) / 10000000L);
+	tp->tv_usec = (long)(system_time.wMilliseconds * 1000);
+	return 0;
+}
 
+#else
+#include <sys/time.h>
+#endif
 
 using namespace std;
 
+#define USENVML
+
+#if defined(USENVML)
+#include <nvml.h>
+bool nv_log_pwr = false;
```

Comment:
I have some personal objections against global variables but that might be just because of their drawbacks in Julia :wink:

Reply:
It seemed like the easiest option to implement shared memory for the std::threads without having to define a specific class. I agree, that it is not the most elegant solution. Any better ideas?

Reply:
Local variables and a lambda/closure? You can also use global variables if you like. I won't reject your changes because of that :smile:

Reply:
Maybe we could use global variables for now and if all works well, wrap everything in a seperate class with correct non global variables for shared memory when we have time for cleanup. It should be done at some point anyways, otherwise the main.cpp will get overcrowded, especially if we also implement things like  Intel® Power Gadget.